docs/ibc/1_IBC_ARCHITECTURE.md

```diff
@@ -0,0 +1,141 @@
+
+### Connections
+
+Reliability, flow control, and authentication as described above require that IBC initializes and maintains certain status information for each datastream. This information is called a connection. Each connection object contains information about the consensus state of the connected ledger, negotiated encoding & multiplexing options, and state & sequence numbers. When two modules wish to communicate, they must locate an existing connection between their two ledgers, or initialize a new connection if none yet exists. Initializing a connection requires a multi-step handshake which, once complete, ensures that only the two intended ledgers are connected and future datagrams relayed will be authenticated, encoded, and sequenced as desired.
+
+### Multiplexing
+
+To allow for many modules within a single host ledger to use an IBC connection simultaneously, IBC provides a set of channels within each connection, which each uniquely identify a linear datapipe over which packets can be sent in order to a destination module on the receiving ledger. Channels are usually expected to be associated with a single module on each ledger, but one-to-many and many-to-one channels are also possible. The number of channels is unbounded, facilitating concurrent throughput limited only by the throughput of the underyling ledgers with only a single connection necessary to track consensus information (and consensus transcript verification cost thus amortized across all channels using the connection).
+
+## Dataflow
+
+IBC can be conceptualized as a layered protocol stack, through which data flows top-to-bottom (when sending IBC packets) and bottom-to-top (when receiving IBC packets).
+
+Consider the path of an IBC packet between two chains — call them *A* and *B*:
+
+### Diagram
+
+```
++----------------------------------------------------------------------------------+
+| Chain A                                                                          |
+|                                                                                  |
+| Actor --> Module --> Handler --> Packet --> Channel --> Connection --> Consensus |
++----------------------------------------------------------------------------------+
+
+    +---------+
+==> | Relayer | ==>
+    +---------+
+
++----------------------------------------------------------------------------------+
+| Chain B                                                                          |
+|                                                                                  |
+| --> Consensus --> Connection --> Channel --> Packet --> Handler --> Module       |
++----------------------------------------------------------------------------------+
+```
+
+### Steps
+
+1. On chain *A*
+    1. Actor (application-specific)
+    1. Module (application-specific)
+    1. Handler (parts defined in different ICSs)
```

Comment:
What is the handler in this context?

Reply:
The part of the state machine implementing the IBC protocol, which routes calls from modules to channels. Added a line in the section synopsis.